mesheryctl/cmd/init.go

```diff
@@ -32,13 +36,33 @@ var initCmd = &cobra.Command{
 			log.Info("Prerequisite Docker Compose not available. Attempting Docker Compose installation...")
 			ostype, osarch := prereq()
 			osdetails := strings.TrimRight(string(ostype), "\r\n") + "-" + strings.TrimRight(string(osarch), "\r\n")
-			dockerComposeBinaryURL := dockerComposeBinaryURL + "-" + osdetails
+
+			//checks for the latest docker-compose
+			resp, err := http.Get("https://api.github.com/repos/docker/compose/releases/latest")
+			dockerComposeBinaryURL := dockerComposeBinaryURL
+			if err != nil {
+				// download the default version as 1.24.1 if unable to fetch newest page data
+				dockerComposeBinaryURL = dockerComposeBinaryURL + "1.24.1/docker-compose"
+			} else {
+				var dat map[string]interface{}
+				body, _ := ioutil.ReadAll(resp.Body)
```

Comment:
I think it will be good to do a response status code check before you try to parse the body.